File: fe/src/lib/VaultSDK/vault/webflow.ts

import Base, { Result } from "../base";
import utils from "../utils";

export interface WebflowAuthResponse  {
    url: string;
}

export interface WebflowTokenResponse {
    token: string;
    sites: WebflowSite[];
}

export interface WebflowSitesResponse {
    sites: WebflowSite[];
}

export interface WebflowSite {
    id : string;
    displayName : string;
    shortName: string;
    customDomains: WebflowDomain[]; 
}

export interface WebflowDomain {
    id : string;
    url: string;
}


export interface WebflowPagesResponse {
    pages: WebflowPage[];
}

export interface WebflowPage {
    id : string;
    title : string;
    slug: string;
    parentId: string; 
    path: string; 
}


export class VaultWebflow extends Base {

    async getWebflowAuthorizeUrl(): Promise<WebflowAuthResponse> {
        //const header = { "x-session-token": token };
        return await utils.http.get(
            this._baseURL,
            `/webflow/auth`,
            {},
            undefined
        );
    }

    async getWebflowSites(token: string): Promise<WebflowSitesResponse> {
        //const header = { "x-session-token": token };
        return await utils.http.post(
            this._baseURL,
            `/webflow/sites`,
            {token: token},
            undefined
        );
    }

    async getWebflowPages(token: string, siteid: string): Promise<WebflowPagesResponse> {
        return await utils.http.post(
            this._baseURL,
            `/webflow/pages`,
            {token: token, site_id: siteid},
            undefined
        );
    }

    async getWebflowToken(code: string): Promise<WebflowTokenResponse> {
        //const header = { "x-session-token": token };
        return await utils.http.post(
            this._baseURL,
            `/webflow/token`,
            {code: code},
            undefined
        );
    }


    async uploadWebflowScript ( token : string, siteId : string, app_id: string) : Promise<boolean> {
        return await utils.http.post(
            this._baseURL,
            `/webflow/upload`,
            {token: token, site_id: siteId, app_id: app_id},
            undefined
        );
    }
}

File: api/services/webflow/webflow.go
package webflow

type WebflowSitesResult struct {
	Sites []WebflowSite `json:"sites"`
}

type WebflowSite struct {
	ID            string          `json:"id"`
	DisplayName   string          `json:"displayName"`
	ShortName     string          `json:"shortName"`
	CustomDomains []WebflowDomain `json:"customDomains"`
}

type WebflowPagesResult struct {
	Pages []WebflowPage `json:"pages"`
}

type WebflowPage struct {
	ID       string `json:"id"`
	ParentID string `json:"parentId"`
	Title    string `json:"title"`
	Slug     string `json:"slug"`
	Path     string `json:"path"`
}

type WebflowDomain struct {
	ID  string `json:"id"`
	Url string `json:"url"`
}

type WebflowRegisteredScriptsResult struct {
	RegisteredScripts []WebflowRegisteredScript `json:"registeredScripts"`
}

type WebflowRegisteredScript struct {
	ID          string `json:"id"`
	Version     string `json:"version"`
	DisplayName string `json:"displayName"`
}

type WebflowAddScriptsRequest struct {
	Scripts []WebflowAddScript `json:"scripts"`
}

type WebflowAddScript struct {
	ID       string `json:"id"`
	Version  string `json:"version"`
	Location string `json:"location"`
}

type WebflowAddScriptsResult struct {
	Scripts []WebflowAddScript `json:"scripts"`
}

type WebflowButtonTemplate struct {
	WalletURL string
	AppID     string
}

type WebflowProtectedPagesTemplate struct {
	WalletApiURL   string
	AppID          string
	PublicKey      string
	ProtectedPages string
	LoginPage      string
}

type WebflowSettings struct {
	SiteID         string        `json:"site_id"`
	SiteName       string        `json:"site_name"`
	SiteShortName  string        `json:"site_shortname"`
	LoginPage      string        `json:"login_page"`
	ProjectedPages []WebflowPage `json:"protected_pages"`
}
